python: Don't strip whitespace from build values from cmake.

arthurp opened the pull request:
Trailing whitespace in LINKER_WRAPPER_FLAG is significant. See
https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_LINKER_WRAPPER_FLAG.html

ddn0 commented:
@insertinterestingnamehere if no one is using `make_dist.sh` for releases, we can also remove `make_dist.sh.in` and the related CMake machinery (scripts/CMakeLists.txt).

roshandathathri commented:
I think we are preemptively removing some scripts which may turnout be useful. "iss_load_modules" is one such example. I created most of the heterogeneousGalois scripts and I use them, especially to run on Stampede/Bridges. Some of the log parser scripts also fall in this category. These are "Galois" scripts, not "Katana" scripts but it is easy to port them. 

If we intend to separate out end-user and developer scripts, we should have separate sub-folders for them, instead of deleting all developer scripts.

I'm definitely not in favor deleting vtune, plot, and log-parser scripts.

ddn0 commented:
I think the current set of removals is conservative, so let's just concentrate on removing these now.

I understand that there might be some outstanding uses, but I don't think the negative impact is that high considering that ISS/Galois still exists. In general, having untested, out-of-date or unmaintained code is a tarpit for developers. The best way to keep something in the codebase is to have a plan for its continual maintenance.

gurbinder533 commented:
@roshandathathri I have tried to remove scripts that I don't see being used for katana in near future (for example job scheduling scripts on Stampede/Bridges) and keeping them around without maintaining is not good. And as @ddn0 said we have a backup for these in ISS/Galois anyways.

roshandathathri commented:
> I don't think the negative impact is that high considering that ISS/Galois still exists

Yes, so, it's fine. 

> I don't see being used for katana in near future

I agree that this is a good rubric.

gurbinder533 commented:
I think we have a consensus then. I will merge this in some time. Let me know if that is not the case.

l-hoang commented:
For the record, I'm on the side of "remove everything that doesn't have to do with Katana" as mentioned earlier, but merging this in is fine for now.

arthurp committed:
python: Don't strip whitespace from build values from cmake.

Trailing whitespace in LINKER_WRAPPER_FLAG is significant. See
https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_LINKER_WRAPPER_FLAG.html
